.travis.yml

```diff
@@ -4,8 +4,6 @@ services:
   - docker
 
 before_install:
-  # make comparing to origin/master work and fetch full history for correct current and previous version detection
-  - git remote set-branches --add origin master && git fetch --unshallow
```

Comment:
This is required (as the comment states) as dynver and MiMa can get problems without it (Only shows when there is a lot of commits between versions).

Reply:
This was failing. That's why I removed. I guess there are some pre-conditions in the repo before you can call `fetch --unshallow`. 

I still need to investigate it. Maybe @dwijnand can shed some light on it.

Reply:
```
$ git clone --depth=50 https://github.com/akka/akka-projection.git akka/akka-projection
...
$ git remote set-branches --add origin master && git fetch --unshallow
fatal: --unshallow on a complete repository does not make sense
The command "git remote set-branches --add origin master && git fetch --unshallow" failed and exited with 128 during .
```

You can make that

```bash
[ $(git rev-parse --is-shallow-repository) = true ] && git fetch --unshallow
```

Reply:
Try to use Travis settings:
```
git:
  depth: false # Avoid sbt-dynver not seeing the tag
```

Reply:
@ihostage, are you saying what this travis option makes the `git fetch --unshallow` redundant?

Reply:
To be honest, I think I prefer: 
`[ $(git rev-parse --is-shallow-repository) = true ] && git fetch --unshallow`

Simply because if depends on git and not some Travis impl.

Reply:
It says Travis itself 😄
https://docs.travis-ci.com/user/customizing-the-build#git-clone-depth
> Some operations on the repository, such as common automated code review scripts (e.g. Pronto for Ruby), may fail due to the limited git clone depth, not being able to access all the objects in the repository. Removing the depth flag, or running `git fetch --unshallow` might solve the issue.